File: easy/binary-search/recursive.c
int search(int* nums, int numsSize, int target){
    return bs(nums, 0, numsSize, target);
}

int bs(int *arr, int low, int high, int target) {
    if (low > high)
        return -1;

    int mid = (low + high) / 2;

    if (arr[mid] == target)
        return mid;
    else if (arr[mid] > target)
        return bs(arr, low, mid - 1, target);
    else // if (arr[mid] < target)
        return bs(arr, mid + 1, high, target);
}

File: easy/binary-search/recursive.py
class Solution:
    def search(self, nums: List[int], target: int) -> int:
        return aux(nums, target, 0, len(nums) - 1)

def aux(nums: List[int], target: int, low: int, high: int) -> int:
    if low <= high:
        mid = (high - low) // 2 + low
        if target > nums[mid]:
            return aux(nums, target, mid + 1, high)
        elif target < nums[mid]:
            return aux(nums, target, low, mid - 1)
        else:
            return mid
    else:
        return -1
